### PR TITLE
fix: working setup-python@v4 with /opt cache

### PR DESCRIPTION
**Pull request by nvonahsen**
Ok this time it's actually tested and working.
Looks like just bumping the setup-python to the latest major did the trick.

### CODE REVIEW

File: .github/workflows/python-pr-cml.yaml [hunk 1]
```diff
@@ -116,9 +116,13 @@ jobs:
           node-version: 16
 
       - name: Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.9
+        env:
+          # Workaround for setup-python: https://github.com/actions/setup-python/issues/485
+          AGENT_TOOLSDIRECTORY: "/opt/hostedtoolcache"
```
Comment (danieljimeneznz) [inline .github/workflows/python-pr-cml.yaml:124]: Do we still need this?

**Review by danieljimeneznz — APPROVED** [review]

File: .github/workflows/python-pr-cml.yaml (hunk 1 above)
Comment (nvonahsen) [reply .github/workflows/python-pr-cml.yaml:124]: Yup

Comment (nvonahsen) [reply .github/workflows/python-pr-cml.yaml:124]: That is still the fix for the thing that broke (which I think is that github moved the tools cache directory or something and that's not being handled properly at some point down the line)

Its just that this env broken something else in v2, whereas in v4 it doesn't.

**Comment by nvonahsen** [comment]
For reference I also did a bunch more digging. The underlying error seems to be that the `gcc` command which is used to build the package still gets `/opt/hostedtoolcache` passed to it's `-I` includes so it doesn't find python which now lives in `cml_dir/_work/_tool`.

This fix doesn't fix that underlying issue, but instead moves where python is installed back to `/opt/hostedtoolcache`.

I did try to figure out how to add the new path to the includes but adding it to `$CMAKE` doesn't work and there is no way to more directly pass it to the build when using poetry.

I still have no idea where the `/opt/` dir is coming from in the `gcc` args tho.

**Pull request merged**